### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by josuerushanika**
In this PR i did :

- [ ] Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
- [ ] Set up associations between models. Remember that author_id column in posts table should be the foreign_key for the users table.
- [ ] Use models to insert data Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- [ ] Create custom methods
 models should include some additional methods.

**_Users_**
A method that returns the 3 most recent posts for a given user.
**_Posts_**
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
**_Comments_**
A method that updates the comments counter for a post.
**_Likes_**
A method that updates the likes counter for a post.
Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  def recent_posts
```
Comment (brytebee) [inline app/models/post.rb]: Kindly rename this method to `recent_comments` get the last 5 comments and not posts.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @josuerushanika 👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB setup.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (peterrobert) [reply app/models/post.rb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @josuerushanika, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by josuerushanika** [comment]
@peterrobert  Thank you

**Pull request merged**